### ISSUE TITLE
Broken typings for typescript

### ISSUE THREAD

**Issue by andyryan** [HubSpot/hubspot-api-nodejs#466]
It is not possible to follow along with documentation as it seems a number if enums, interfaces and types are not being correctly exported. 

Ass an example, I am trying to create an association as listed in the documentation 

```
 this.client.crm.associations.v4.basicApi.create(
                'contacts', contact.id, 'companies', associatedCompany.id, [
                    {
                        associationCategory: 'HUBSPOT_DEFINED',
                        associationTypeId: Hubspot.AssociationTypes.contactToCompany
                    }
                ])
```

Where hubspot is imported with 
```
import * as Hubspot from '@hubspot/api-client'
```
 and 
```
this.client = new Hubspot.Client({...}}
```

There is a typescript linting error on 'HUBSPOT_DEFINED':
> Type '"HUBSPOT_DEFINED"' is not assignable to type 'AssociationSpecAssociationCategoryEnum'.

This is not the only place this has occured. When trying to use the search API as instructed buy the documentation, the `sorts` property on the search request requires type `[string]` where the documentation lists an object.

At the moment our only solution is to make the file utilising this module a JS file

**Comment by andyryan** [comment]
This only appears to be an issue since 11.0. Rolling back to 10.2 has removed these error messages.

**Comment by oschwede** [comment]
We also experienced this. The reason is, that some properties have been converted from union types to enums. Unfortunately, string enums behave a bit differently:
```ts
type A = "FOO" | "BAR";
enum B { FOO ="FOO", BAR = "BAR" }

const a: A = "FOO"; // works
const b: B = "FOO"; // Does not work anymore
```

The latter is the problem that has been introduced here. It seems to be intended behavior by typescript, see https://github.com/microsoft/TypeScript/issues/17690

Exporting the enums would fix this issue

**Comment by klu2** [comment]
Same here, we have the same issue with the `FilterOperatorEnum` which has been introduced, but there is another breaking change as well that I would like to put into this ticket

we previously had

```typescript
        const existingProduct = await this.hubspotClient.crm.products.searchApi.doSearch({
            filterGroups: [
                {
                    filters: [
                        {
                            propertyName: 'internal_product_id',
                            operator: 'EQ',
                            value: orderItem.internalProductId.toString(),
                        },
                    ],
                },
            ],
            sorts: ['internal_product_id'],
            properties: ['hs_product_id', 'internal_product_id'],
            limit: 1,
            after: 0,
        });
```

We can change the `filter` to the following code:

```typescript
filters: [
  {
     propertyName: 'internal_product_id',     
     operator: FilterOperatorEnum.Eq,
     value: orderItem.internalProductId.toString(),
  },
],
```

And I do appreciate the strict typing with those enums instead of having string constants, but the problem is that the parameter `after` also changed from `number` to `string`, I created another ticket (https://github.com/HubSpot/hubspot-api-nodejs/issues/467) to clarify that

**Comment by klu2** [comment]
Still there is one more issue I'm having here with the enums AssocitionCategories that has another reason I believe, so I'd like to comment it here:

What we did with 10.x.x was something like that to fetch an association from Hubspot:

```typescript
const definitions = await this.hubspotClient.crm.associations.v4.schema.definitionsApi.getAll(
    'contact',
    'contact',
);
// in reality we did some filtering here based on labels to fetch the right object, here we are taking the first
let myAssociation = definitions.results[0];
```

`myAssoctiation` is a `AssociationSpecWithLabel`.

Then I wanted to take that association spec to create a new association:

```typescript
 await this.hubspotClient.crm.associations.v4.basicApi.create(
    'contact',
    result.id,
    'contact',
    upline.hubspotContactId,
        [
            {
                associationTypeId: myAssoctiation.typeId,
                associationCategory: myAssoctiation.category,
            },
       ],
;
```
The problem now is that `myAssocation.category` is a `AssociationSpecWithLabelCategoryEnum` but `associationCategory` expects a `AssociationSpecAssociationCategoryEnum`

**Comment by ksvirkou-hubspot** [comment]
Hi everyone.
All enums have been converted from union types to enums in new version(11.0.0-beta). [Link to changelog.](https://github.com/HubSpot/hubspot-api-nodejs/blob/master/CHANGELOG.md?plain=1#L62)
This change was initiated by the community in [this issue](https://github.com/HubSpot/hubspot-api-nodejs/issues/172).
All examples in the README were wrote on JS, I'll make a note about it.

**Comment by klu2** [comment]
The majority of points works great so far with the new release, but my point with `AssociationSpecWithLabelCategoryEnum` and `AssociationSpecAssociationCategoryEnum` is still a case

**Comment by klu2** [comment]
Still, I found one more binding issue. `MultiAssociatedObjectWithLabel.toObjectId` is marked as a `string` in `MultiAssociatedObjectWithLabel.d.ts` but if you fetch such an object via `client.crm.associations.v4.basicApi.getPage`, this field is a `number` as runtime. 

So it seems that one binding there is not yet correct. Shall I raise another issue for that or do you want to reopen that one?

**Comment by ksvirkou-hubspot** [comment]
All object IDs in the SDK have to be `string`. 
It was done for user can get objects by ids getting from associations API without converting.

**Comment by ksvirkou-hubspot** [comment]
> Still, I found one more binding issue. `MultiAssociatedObjectWithLabel.toObjectId` is marked as a `string` in `MultiAssociatedObjectWithLabel.d.ts` but if you fetch such an object via `client.crm.associations.v4.basicApi.getPage`, this field is a `number` as runtime.
> 
> So it seems that one binding there is not yet correct. Shall I raise another issue for that or do you want to reopen that one?

The problem is on API side, because all object IDs have to be string.
It was fixed on the SDK side for getting objects by ids from associations API without converting.
Any issue by this cause?

**Comment by ksvirkou-hubspot** [comment]
> The majority of points works great so far with the new release - thanks for that! -, but my point with `AssociationSpecWithLabelCategoryEnum` and `AssociationSpecAssociationCategoryEnum` is still a case
> 
> What I did as a workaround now is to convert one enum internally to another:
> 
> ```ts
>     private convertCategory(category: AssociationSpecWithLabelCategoryEnum): AssociationSpecAssociationCategoryEnum {
>         switch (category) {
>             case AssociationSpecWithLabelCategoryEnum.HubspotDefined:
>                 return AssociationSpecAssociationCategoryEnum.HubspotDefined;
>             case AssociationSpecWithLabelCategoryEnum.UserDefined:
>                 return AssociationSpecAssociationCategoryEnum.UserDefined;
>             case AssociationSpecWithLabelCategoryEnum.IntegratorDefined:
>                 return AssociationSpecAssociationCategoryEnum.IntegratorDefined;
>         }
>     }
> ```

You can open another issue and I'll try to do something with that.
